mm2src/trezor/src/utxo/utxo_command.rs

```diff
@@ -0,0 +1,69 @@
+use crate::client::TrezorClient;
+use crate::constants::TrezorCoin;
+use crate::proto::messages_bitcoin as proto_bitcoin;
+use crate::response_channel::{response_loop, trezor_response_channel, TrezorResponseReceiver};
+use crate::{ecdsa_curve_to_string, serialize_derivation_path, TrezorResult};
+use common::executor::spawn;
+use futures::future::{select, FutureExt};
+use hw_common::primitives::{DerivationPath, EcdsaCurve};
+
+// Bitcoin(UTXO) operations.
+impl TrezorClient {
+    pub fn get_utxo_address(
+        &self,
+        path: DerivationPath,
+        coin: TrezorCoin,
+        show_display: bool,
+    ) -> TrezorResponseReceiver<TrezorResult<String>> {
+        let trezor = self.clone();
+        let (response_tx, response_rx, shutdown_rx) = trezor_response_channel();
+        let fut = async move {
+            let req = proto_bitcoin::GetAddress {
+                address_n: serialize_derivation_path(&path),
+                coin_name: Some(coin.to_string()),
+                show_display: Some(show_display),
+                multisig: None,
+                script_type: None,
+                ignore_xpub_magic: None,
+            };
+
+            let result_handler = Box::new(|m: proto_bitcoin::Address| Ok(m.address));
+            let result = trezor.call(req, result_handler).await;
+            response_loop(response_tx, result).await;
```

Comment:
This is what I still puzzle over. Please look at this function:
https://github.com/KomodoPlatform/atomicDEX-API/blob/c032ff4391fa7533781055b7adbe009152412930/mm2src/crypto/src/crypto_ctx.rs#L131-L133

Wrapping the response into a stream allows us to use chain operations like `and_then`. As for now, it looks very complicated, because the solution consists of legacy `TrezorResponse` implementation and `TrezorResponseReceiver`.

But now I'm trying to figure out how to refactor it, make it easier and thread-safety. It seems like the only way to avoid using `and_then` at all. Then `CryptoCtx::init_with_trezor` will look like:
```rust
pub async fn init_with_trezor(ctx: MmArc, task_handle: &RpcTaskHandle) -> CryptoInitResult<()>
```

instead of
```rust
pub async fn init_with_trezor(ctx: MmArc) -> TrezorResponseReceiver<CryptoInitResult<()>>
```

Reply:
Was just asking :slightly_smiling_face: Both solutions look ok to me on the current iteration so it's not a blocker.

Reply:
There is no wrapping the response into a stream more :D
Could you please take a look at this?
Also this PR is ready for a final review